### DISCUSSION TITLE
openRerity core algorithm JS implementation

### DISCUSSION THREAD

**Discussion by kongnet** [OpenRarity/open-rarity#61] [Ideas]
Openrarity is great, I learned and implemented part of the algorithm with JS.
[CODE](https://github.com/kongnet/openrarity)

**Comment by impreso** [comment]
Thanks , i will take a look! Do you consider Null traits? ( Probabilities that attribute is not in token)

**Reply by kongnet** [reply]
Null traits is already include. My step is
1.Generate trait value map without undefined(blank) trait
2.Generate undefined(blank) trait for every token
3.Generate trait value stats include traitEntyrop
4.Generate token trait probability list

And add the same python Testing in JS Testing

**Comment by impreso** [comment]
Hi , we are working on official OpenRarity TypeScript support.

**Reply by theboxer** [reply]
I'd like to offer a help with the TS support

**Comment by impreso** [comment]
Will close this thread for now - stay tuned on TS implementation.
